Snapdragon X Elite Arm / Windows , No native build was found for platform=win32 arch=arm64

### Which @angular/* package(s) are the source of the bug?

compiler, core, Don't known / other

### Is this a regression?

Yes

### Description

 I tried Angular 18 on my  new Snapdragon X Elite

When I try to run the project from VSCode I get this error. 
No native build was found for platform=win32 arch=arm64 runtime=node abi=127 uv=1 armv=8 libc=glibc node=22.3.0
    attempted loading from: D:\Junk\my-app\node_modules\lmdb and package: @lmdb/lmdb-win32-arm64
 

### Please provide a link to a minimal reproduction of the bug

_No response_

### Please provide the exception or error you saw

```true
ng serve --o
Application bundle generation failed. [0.406 seconds]

X [ERROR] No native build was found for platform=win32 arch=arm64 runtime=node abi=127 uv=1 armv=8 libc=glibc node=22.3.0
    attempted loading from: D:\Junk\my-app\node_modules\lmdb and package: @lmdb/lmdb-win32-arm64
 [plugin angular-compiler]
```


### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 18.0.4
Node: 22.3.0
Package Manager: npm 10.8.1
OS: win32 arm64

Angular: 18.0.3
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1800.4
@angular-devkit/build-angular   18.0.4
@angular-devkit/core            18.0.4
@angular-devkit/schematics      18.0.4
@angular/cli                    18.0.4
@schematics/angular             18.0.4
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.7
```


### Anything else?

React JS works Fine on the same platform

--- a/packages/angular/build/package.json
+++ b/packages/angular/build/package.json
@@ -33,7 +33,6 @@
     "https-proxy-agent": "7.0.5",
     "istanbul-lib-instrument": "6.0.3",
     "listr2": "8.2.5",
-    "lmdb": "3.1.3",
     "magic-string": "0.30.11",
     "mrmime": "2.0.0",
     "parse5-html-rewriting-stream": "7.0.0",
@@ -44,6 +43,9 @@
     "semver": "7.6.3",
     "vite": "5.4.8",
     "watchpack": "2.4.2"
+  },
+  "optionalDependencies": {
+    "lmdb": "3.1.3"
   },
   "peerDependencies": {
     "@angular/compiler": "^19.0.0-next.0",

